pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/SimpleConsumerWrapper.java

```diff
@@ -333,21 +334,25 @@ private void setCurrentState(State newState) {
     _currentState = newState;
   }
 
-  public synchronized int getPartitionCount(String topic, long maxWaitTimeMs) {
+  public synchronized int getPartitionCount(String topic, long timeoutMillis) {
     int unknownTopicReplyCount = 0;
     final int MAX_UNKNOWN_TOPIC_REPLY_COUNT = 10;
     int kafkaErrorCount = 0;
     final int MAX_KAFKA_ERROR_COUNT = 10;
 
-    while(true) {
+    final long endTime = System.currentTimeMillis() + timeoutMillis;
+
+    while(System.currentTimeMillis() < endTime) {
       // Try to get into a state where we're connected to Kafka
-      // TODO This needs a time limit, and perhaps a back-off before we query kafka again.
-      while (!_currentState.isConnectedToKafkaBroker()) {
+      while (!_currentState.isConnectedToKafkaBroker() && System.currentTimeMillis() < endTime) {
         _currentState.process();
       }
 
-      // Send the metadata request to Kafka
+      if (endTime <= System.currentTimeMillis() && !_currentState.isConnectedToKafkaBroker()) {
+        throw new TimeoutException();
```

Comment:
and partition number. Perhaps it is best to instantiate a separate logger like we have done in LLSegmentDataManager that has topic + partition in it?